repobee_sanitizer/_sanitize_repo.py

```diff
@@ -114,8 +134,22 @@ def _discover_dirty_files(
 
 def _sanitize_files(
     basedir: pathlib.Path, file_relpaths: List[_fileutils.RelativePath]
-) -> None:
-    """Sanitize the provided files."""
+) -> Optional[_FilesWithErrors]:
+    """Checks the syntax of the provided files and reports any found errors.
+    If any errors are found, report errors and exits. If there are no errors,
+    then all files are sanitized."""
+    files_with_errors = []
+
+    for relpath in file_relpaths:
+        file_abspath = basedir / str(relpath)
+
+        errors = _syntax.check_syntax(file_abspath.read_text().split("\n"))
+        if errors:
+            files_with_errors.append(_FilesWithErrors(relpath, errors))
+
+    if files_with_errors:
+        return files_with_errors
```

Comment:
Just return the list. If it's empty, there are no errors. As a general advice, avoid returning `None` if possible.

```suggestion
    return files_with_errors
```

Reply:
If we remove the if statement the code will not go on to the next loop